react/features/video-layout/subscriber.js

```diff
@@ -4,21 +4,69 @@ import {
     VIDEO_QUALITY_LEVELS,
     setMaxReceiverVideoQuality
 } from '../base/conference';
+import {
+    getPinnedParticipant,
+    pinParticipant
+} from '../base/participants';
 import { StateListenerRegistry } from '../base/redux';
 import { selectParticipant } from '../large-video';
 import { shouldDisplayTileView } from './functions';
 
+declare var interfaceConfig: Object;
+
 /**
  * StateListenerRegistry provides a reliable way of detecting changes to
  * preferred layout state and dispatching additional actions.
  */
 StateListenerRegistry.register(
     /* selector */ state => shouldDisplayTileView(state),
-    /* listener */ (displayTileView, { dispatch }) => {
-        dispatch(selectParticipant());
+    /* listener */ (displayTileView, store) => {
+        store.dispatch(selectParticipant());
 
         if (!displayTileView) {
-            dispatch(setMaxReceiverVideoQuality(VIDEO_QUALITY_LEVELS.HIGH));
+            store.dispatch(
+                setMaxReceiverVideoQuality(VIDEO_QUALITY_LEVELS.HIGH));
+
+            _updateAutoPinnedParticipant(store);
         }
     }
 );
+
+/**
+ * For auto-pin mode, listen for changes to the last known screen share
+ * participant and automatically pin that participant.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => state['features/video-layout'].screenShares,
+    /* listener */ (screenShares, store) =>
+        _updateAutoPinnedParticipant(store));
+
+
+/**
+ * Private helper to automatically pin the latest screen share stream or unpin
+ * if there are no more screen share streams.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {void}
+ */
+function _updateAutoPinnedParticipant({ dispatch, getState }) {
+    if (typeof interfaceConfig !== 'object'
+            || !interfaceConfig.AUTO_PIN_LATEST_SCREEN_SHARE) {
+        return;
+    }
+
+    const state = getState();
+    const screenShares = state['features/video-layout'].screenShares;
+
+    if (!screenShares) {
+        return;
+    }
+
+    const latestScreenshareParticipantId = Array.from(screenShares).pop();
```

Comment:
Isn't it going to be better to store the screen shares as Array?

Reply:
JS Sets preserve order. I can look around for explicit documentation that Array.from would preserve the same order.

Reply:
Sets have look methods that are more convenient to use versus iteration over the entire array. That said, the array of screenshares should be short so any perf gain would be negated and maybe non existent with short lengths. I moved to an array because I'm not strongly for the Set.

Reply:
Since the `screenShares` is `Array` now why do we still need `Array.from`?

Reply:
@virtuacoplenny I think here you shouldn't be worried of modifying the `screenShares`. But if you are worried isn't it better to use `screenShares[screenShares.length-1]`?